pkg/autoscaler/stats_reporter_test.go

```diff
@@ -73,58 +73,58 @@ func TestReporter_Report(t *testing.T) {
+	expectSuccess(t, "ReportRequestedPodCount", func() error { return r.ReportRequestedPodCount(5) })
+	expectSuccess(t, "ReportRequestedPodCount", func() error { return r.ReportRequestedPodCount(6) })
 	checkData(t, "requested_pods", wantTags, 6)
 
-	expectSuccess(t, func() error { return r.Report(ActualPodCountM, 7) })
-	expectSuccess(t, func() error { return r.Report(ActualPodCountM, 8) })
-	expectSuccess(t, func() error { return r.Report(ActualPodCountM, 9) })
+	expectSuccess(t, "ReportActualPodCount", func() error { return r.ReportActualPodCount(7) })
+	expectSuccess(t, "ReportActualPodCount", func() error { return r.ReportActualPodCount(8) })
+	expectSuccess(t, "ReportActualPodCount", func() error { return r.ReportActualPodCount(9) })
 	checkData(t, "actual_pods", wantTags, 9)
 
-	expectSuccess(t, func() error { return r.Report(PanicM, 1) })
-	expectSuccess(t, func() error { return r.Report(PanicM, 0) })
-	expectSuccess(t, func() error { return r.Report(PanicM, 1) })
+	expectSuccess(t, "ReportPanic", func() error { return r.ReportPanic(1) })
+	expectSuccess(t, "ReportPanic", func() error { return r.ReportPanic(0) })
+	expectSuccess(t, "ReportPanic", func() error { return r.ReportPanic(1) })
 	checkData(t, "panic_mode", wantTags, 1)
 
-	expectSuccess(t, func() error { return r.Report(PanicM, 0) })
+	expectSuccess(t, "ReportPanic", func() error { return r.ReportPanic(0) })
 	checkData(t, "panic_mode", wantTags, 0)
 }
 
-func expectSuccess(t *testing.T, f func() error) {
+func expectSuccess(t *testing.T, funcName string, f func() error) {
 	if err := f(); err != nil {
-		t.Errorf("Reporter.Report() expected success but got error %v", err)
+		t.Errorf("Reporter.%v() expected success but got error %v", funcName, err)
 	}
 }
 
 func checkData(t *testing.T, name string, wantTags map[string]string, wantValue float64) {
 	if d, err := view.RetrieveData(name); err != nil {
-		t.Errorf("Reporter.Report() error = %v, wantErr %v", err, false)
+		t.Errorf("Got unexpected errro from view.RetrieveData error: %v", err)
 	} else {
 		if len(d) != 1 {
-			t.Errorf("Reporter.Report() len(d) %v, want %v", len(d), 1)
+			t.Errorf("want 1 data row but got %v", len(d))
 		}
 		for _, got := range d[0].Tags {
 			if want, ok := wantTags[got.Key.Name()]; !ok {
-				t.Errorf("Reporter.Report() got an extra tag %v: %v", got.Key.Name(), got.Value)
+				t.Errorf("got an extra tag %v: %v", got.Key.Name(), got.Value)
 			} else {
 				if got.Value != want {
-					t.Errorf("Reporter.Report() expected a different tag value. key:%v, got: %v, want: %v", got.Key.Name(), got.Value, want)
+					t.Errorf("expected a different tag value. key:%v, got: %v, want: %v", got.Key.Name(), got.Value, want)
 				}
 			}
 		}
 
 		if s, ok := d[0].Data.(*view.LastValueData); !ok {
-			t.Error("Reporter.Report() expected a LastValueData type")
+			t.Error("expected a LastValueData type")
 		} else {
 			if s.Value != (float64)(wantValue) {
```

Comment:
I guess, it's fine for the test, but in general you don't compare floats as ==, since it might generate false results.

Reply:
Ack.